clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerSaslClient.java

```diff
@@ -150,6 +154,20 @@ private void setState(State state) {
         this.state = state;
     }
 
+    private SaslExtensionsCallback parseCustomExtensions() throws SaslException {
```

Comment:
Is there a reason to return the Callback?   I would think the calling code is interested in the extensions themselves rather than the callback.

Reply:
You're right, even the method name implies you get the extensions